### DISCUSSION TITLE
Feature request - additional shutter mode - 3 pulse relays: 1-open / 2-stop / 3-close

### DISCUSSION THREAD

**Discussion by troda** [arendst/Tasmota#15222] [Ideas]
### PROBLEM DESCRIPTION
Feature request to add new shutter mode to support 3 relays for shutter: one for opening (pulse), one for stopping the movement (pulse), and the last one for closing (pulse).

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [x] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [x] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [x] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [x] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [ ] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [x] Device used (e.g., Sonoff Basic): Sonoff 4CH Pro
- [x] Tasmota binary firmware version number used: 11.0.0.4
  - [x] Pre-compiled
  - [ ] Self-compiled
- [ ] Flashing tools used: _____
- [x] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
  Configuration output here:

{"NAME":"Generic","GPIO":[32,1,1,1,226,225,33,34,224,320,35,227,0,0],"FLAG":0,"BASE":23}
{"Module":{"23":"Sonoff 4CH Pro"}}
{"GPIO0":{"32":"Button1"},"GPIO1":{"0":"None"},"GPIO2":{"0":"None"},"GPIO3":{"0":"None"},"GPIO4":{"226":"Relay3"},"GPIO5":{"225":"Relay2"},"GPIO9":{"33":"Button2"},"GPIO10":{"34":"Button3"},"GPIO12":{"224":"Relay1"},"GPIO13":{"320":"Led_i1"},"GPIO14":{"35":"Button4"},"GPIO15":{"227":"Relay4"},"GPIO16":{"0":"None"},"GPIO17":{"0":"None"}}
```
- [ ] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:


```
- [ ] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:


```
- [ ] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:


```

### TO REPRODUCE
N/A

### EXPECTED BEHAVIOUR
I have shutter (actually awning) which is controller by remote (Somfy system, with Somfy remote). I have connected buttons from the remote to relays of Sonoff 4CH Pro.

Remote has 3 buttons:
(1) To open the shutter (awning) - click/pulse needed.
(2) To stop any ongoing operations - click/pulse needed to stop the movement.
(3) To close the shutter (awning) - click/pulse needed.

I connected all 3 buttons to relays (first to the first relay, second to the second relay, third to the third relay). After that I can use Sonoff 4CH Pro relays to send pulses to control the shutter (awning) - and it works fully.

Now - I would like to configure Tasmota shutter mechanism for this. However there is no shutter mode which could support that.

The closest one is shutter mode 3 - "garage motors" - but it doesn't support stop button and third relay in general.

Would it be possible to add the new mode to support 3 relays connected to control this shutter (awning)? Or maybe it can be easily achieved today?

### SCREENSHOTS
N/A

### ADDITIONAL CONTEXT
N/A

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by cn-superman** [comment]
Indeed, there are many door controllers on the market that use three buttons: open, stop, and close.
![IMG_3722](https://github.com/arendst/Tasmota/assets/45441185/c9371c22-7994-45ee-a012-e406fbcbbb41)

**Reply by stefanbode** [reply]
Yes this is where this mode was created for. Please start with door in the middle an short open and close time of max 5 sec. The. You can confirm everything is correct before using also the endpositons. Your garage should/must have endstops to ensure not to overload the motors.

**Comment by stefanbode** [comment]
I thought it works already but wrong. I assume the 3 relays should trigger the buttons. I think also pressing the up button several times will not change anything. It just continue up. Correct? So to stop you have to use the stop button or wait until the end position is reached.

**Comment by F-rank81** [comment]
I'm looking for just the same thing. As a workaround I added an inverted status led. It's output is always high when there's no moving curtain/shutter/door so it stops.  It does the job but it's not the way I really like to use it. Is there any news on an additional mode?
I hope someone has got a better solution than this...
Thanks to everyone who makes tasmota possible!
It's really awesome :-)

**Comment by stefanbode** [comment]
hi @F-rank81 : Sorry this request got lost and was initially rejected on ESP8266. Never mind on ESP32 we have some more room for options and changes. Submitted: https://github.com/arendst/Tasmota/pull/20758 #20758 
I will update documentation accordingly soon. How does is work:
Define RELAY3 for STOP. Define a PULSETIME on RELAY3
Execute the new command: `shutterStopRelay1 1`
This ENABLE a STOP relay on SHUTTER1. To disable just:
`shutterStopRelay1 0`

**Reply by F-rank81** [reply]
Thanks Stefan.
That looks like it will fit my needs!

Best Regards,
Frank
